packages/react-components/src/components/StreamMedia.tsx

```diff
@@ -69,12 +81,19 @@ export const StreamMedia = (props: StreamMediaProps): JSX.Element => {
   }, [videoStreamElement]);
 
   return (
-    <div
-      className={mergeStyles(
-        isMirrored && pipEnabled ? invertedVideoInPipStyle(theme) : mediaContainer(theme),
-        styles?.root
+    <div className={container()}>
+      <div
+        className={mergeStyles(
+          isMirrored && pipEnabled ? invertedVideoInPipStyle(theme) : mediaContainer(theme),
+          styles?.root
+        )}
+        ref={containerEl}
+      />
+      {isRemoteVideoStream && videoRenderingControls?.remoteVideoStreamLoadingSpinner && !isReceiving && (
```

Comment:
We'd want base components to be as non-opinionated as possible and avoid having complex internal logic that makes it hard for contoso to use -- e.g. if contoso sets `remoteVideoStreamLoadingSpinner = true` without looking at the source code they don't know when a spinner will and won't show as it relies on the exact state of three different props. The goal is for contoso to set explicit props that each impact one and only one thing with high clarity what will happen (for god-like components such as VideoGallery that badly need broken into sub components  this simply isn't possible).

Instead consider a a single prop `showLoadingSpinner` and the parent has to control when this is set to true or false.

We normally also want to avoid booleans in our API as they can't be extended once set (imagine if we want a different kind of loading spinner). So perhaps `showLoadingState?: 'spinner' -- but open to suggestions on naming here (cc @anjulgarg)

Reply:
We have decided to just remove videoRenderingControlOptions